code/datums/traits/negative.dm

```diff
@@ -0,0 +1,110 @@
+//predominantly negative traits
+
+/datum/trait/nearsighted
+	name = "Nearsighted"
+	desc = "You are nearsighted without prescription glasses, but spawn with a pair."
+	value = -1
+	mob_trait = TRAIT_NEARSIGHT
+	gain_text = "<span class='danger'>Things far away from you start looking blurry.</span>"
+	lose_text = "<span class='notice'>You start seeing faraway things normally again.</span>"
+
+/datum/trait/nearsighted/add()
+	if(SSticker.HasRoundStarted()) //this ensures it only creates the glasses when the round starts, never after, because characters are created before the round starts
+		return
+	var/mob/living/carbon/human/H = trait_holder
+	var/obj/item/clothing/glasses/regular/glasses = new(get_turf(H))
+	H.put_in_hands(glasses)
+	H.equip_to_slot(glasses, slot_glasses)
+	H.regenerate_icons() //this is to remove the inhand icon, which persists even if it's not in their hands
+
+
+
+/datum/trait/heavy_sleeper
+	name = "Heavy Sleeper"
+	desc = "You sleep like a rock! Whenever you're put to sleep, you sleep for a little bit longer."
+	value = -1
+	mob_trait = TRAIT_HEAVY_SLEEPER
+	gain_text = "<span class='danger'>You feel sleepy.</span>"
+	lose_text = "<span class='notice'>You feel awake again.</span>"
+
+
+
+/datum/trait/insanity
+	name = "Reality Dissociation Syndrome"
+	desc = "You suffer from a severe disorder that causes very vivid hallucinations. Mindbreaker toxin can suppress its effects."
+	value = -1
+	//no mob trait because it's handled uniquely
+	gain_text = "<span class='userdanger'>...</span>"
+	lose_text = "<span class='notice'>You feel in tune with the world again.</span>"
+
+/datum/trait/insanity/add()
+	addtimer(CALLBACK(src, .proc/disclaimer), 50)
+
+/datum/trait/insanity/process()
+	if(trait_holder.reagents.has_reagent("mindbreaker"))
+		trait_holder.hallucination = 0
+		return
+	if(prob(1)) //we'll all be mad soon enough
+		madness()
+
+/datum/trait/insanity/proc/madness(mad_fools)
```

Comment:
Cmon we have a full hallucination system

Reply:
Do you know where the files for it are? I wanted to put them in that way but I couldn't find them after some searching